### PR TITLE
[PLANNER-2728] Extract optaplanner release

### PR DESCRIPTION
**Pull request by radtriste**
Depends on https://github.com/kiegroup/kogito-pipelines/pull/508

Extracted all optaplanner jobs/dependencies to optaplanner config.

Related: https://github.com/kiegroup/optaplanner/pull/2043

Tests:
- [x] Test DSL generation for Kogito
    - [x] Check no optaplanner job remaining
    - [x] Check trigger jobs configuration
- [x] Test update quarkus all job to update branch config and seed repo PR config correctly
- [x] Test Kogito release job

Remaining:
- [x] Update quarkus platform script to decorrelate kogito and optaplanner updates
- [x] Update GHA to include optaplanner checks

### CODE REVIEW

**Comment by radtriste** [comment]
waiting for https://github.com/kiegroup/kogito-pipelines/pull/508 to be merged before setting that one as ready

File: .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                            // Create PR only if repo/author/branch is different from previous changes
+                            if ("${SEED_REPO}" != "${SEED_BRANCH_CONFIG_FILE_GIT_REPOSITORY}" ||
+                                    "${SEED_AUTHOR_NAME}" != "${SEED_BRANCH_CONFIG_FILE_GIT_AUTHOR_NAME}" ||
+                                    "${SEED_BRANCH}" != "${SEED_BRANCH_CONFIG_FILE_GIT_BRANCH}") {
+                                        String prLink = githubscm.createPR(commitMsg, 'Please review and merge', "${SEED_BRANCH}", "${SEED_AUTHOR_CREDS_ID}")
+                                echo "Created PR ${prLink}"
+
+                                sendNotification("PR was created to update quarkus version in seed repo to ${getNewVersion()}.\nPlease review and merge ${prLink}")
+                            } else {
+                                echo "PR was already created. Will not try to create a new one ..."
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all]:
```suggestion
                                echo 'PR was already created. Will not try to create a new one ...'
```

File: .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                            }
+                        } else {
+                            echo "No changes to commit. No PR is created !"
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all]:
```suggestion
                            echo 'No changes to commit. No PR is created !'
```

File: dsl/config/main.yaml [hunk 3]
```diff
@@ -59,7 +54,7 @@ seed:
       author:
         name: kiegroup
         credentials_id: kie-ci
-      branch: main # Used only for main branch checkout. Else it should be defined for each branch at git.branches level. By default, this value will be overriden by the branch name.
+      branch: main
     path: dsl/config/branch.yaml
   jenkinsfile: dsl/seed/jenkinsfiles/Jenkinsfile.seed.branch
```
Comment (rsynek) [inline dsl/config/main.yaml:59]: The line `optaplanner-quickstarts: development` could probably go away too (sorry, I cannot mark the right line in the review)

**Review by rsynek — APPROVED** [review]
I quickly scanned over the PR and generally, it looks good. Someone more knowledgeable of the Job DSL, seed jobs, and templates should review this PR too.

File: dsl/config/main.yaml (hunk 3 above)
Comment (radtriste) [reply dsl/config/main.yaml:59]: good catch. updated

**Comment by radtriste** [comment]
Moved the GHA part to https://github.com/kiegroup/kogito-pipelines/pull/518

**Review by Ginxo — APPROVED** [review]
LGTM, two minor things

File: .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+                            String commitMsg = "[${SEED_BRANCH_CONFIG_FILE_GIT_BRANCH}] Bump quarkus version to ${getNewVersion()}"
+                            githubscm.commitChanges(commitMsg)
+                            githubscm.pushObject('origin', prBranch, "${SEED_BRANCH_CONFIG_FILE_GIT_AUTHOR_CREDS_ID}")
+
+                            String prLink = githubscm.createPR(commitMsg, 'Please review and merge', "${SEED_BRANCH_CONFIG_FILE_GIT_BRANCH}", "${SEED_BRANCH_CONFIG_FILE_GIT_AUTHOR_CREDS_ID}")
+                            echo "Created PR ${prLink}"
+
+                            sendNotification("PR was created to update quarkus version in branch config to ${getNewVersion()}.\nPlease review and merge ${prLink}")
+                        } else {
+                            echo "No changes to commit. No PR is created !"
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all]:
```suggestion
                            echo 'No changes to commit. No PR is created !'
```

File: .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all (hunk 1 above)
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.ecosystem.update-quarkus-all]:
```suggestion
                                echo 'PR was already created. Will not try to create a new one ...'
```

**Comment by radtriste** [comment]
all tests passed. removed the test commit and all related code change for the testing

**Comment by radtriste** [comment]
DSL failing is expected

**Pull request merged**